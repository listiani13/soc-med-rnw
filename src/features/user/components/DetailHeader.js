// @flow
import React from 'react';
import {View, StyleSheet} from 'react-native';
import {Avatar} from '@material-ui/core';
import {Text} from '../../../core-ui';
import getInitialName from '../../../helpers/getInitialName';
import {GREY} from '../../../constants/color';
import {AVATAR_SIZE} from '../../../constants/size';
import {FONT_SIZE} from '../../../constants/text';

type Props = {
  name: string,
  username: string,
};
export default function DetailHeader(props: Props) {
  let {name, username} = props;
  return (
    <View style={styles.header}>
      <Avatar style={stylesMUI.avatar}>{getInitialName(name)}</Avatar>
      <View style={styles.titleContainer}>
        <Text size="large">{name}</Text>
        <Text style={styles.username}>{username}</Text>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  header: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 10,
  },
  titleContainer: {
    flexDirection: 'column',
  },
  username: {
    color: GREY,
  },
});

const stylesMUI = {
  avatar: {
    width: AVATAR_SIZE,
    height: AVATAR_SIZE,
    marginRight: 25,
    fontSize: FONT_SIZE.xlarge,
  },
};
