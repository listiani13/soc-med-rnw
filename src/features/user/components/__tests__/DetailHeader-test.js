// @flow
import React from 'react';
import {render} from 'react-testing-library';
import DetailHeader from '../DetailHeader';

describe('Detail Header component', () => {
  it('should render Detail Header', () => {
    let {container} = render(
      <DetailHeader name="Name here" username="username" />,
    );
    expect(container).toMatchSnapshot();
  });
});
