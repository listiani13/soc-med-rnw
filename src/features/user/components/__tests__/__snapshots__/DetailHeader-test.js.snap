// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail Header component should render Detail Header 1`] = `
<div>
  <div
    class="rn-alignItems-1awozwy rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-18u37iz rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginLeft-11wrixw rn-marginBottom-15d164r rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
  >
    <div
      class="MuiAvatar-root-1 MuiAvatar-colorDefault-2"
      style="width: 90px; height: 90px; margin-right: 25px; font-size: 36px;"
    >
      NH
    </div>
    <div
      class="rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
    >
      <div
        class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-1471scf rn-fontFamily-14xgk7a rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-lineHeight-t9a87b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
        dir="auto"
        style="font-size: 28px;"
      >
        Name here
      </div>
      <div
        class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-1yis5wz rn-display-1471scf rn-fontFamily-14xgk7a rn-fontSize-1i10wst rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-lineHeight-t9a87b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
        dir="auto"
      >
        username
      </div>
    </div>
  </div>
</div>
`;
